compiler/modules/front/codeGen.bal

```diff
@@ -855,6 +843,96 @@ function codeGenAssignToMember(CodeGenContext cx, bir:BasicBlock startBlock, Env
     }
 }
 
+function codeGenCompoundAssignStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environment env, s:CompoundAssignStmt stmt) returns CodeGenError|StmtEffect {
+    var { lValue, expr , op, pos } = stmt;
+    s:Expr binExpr;
+    if lValue is s:VarRefExpr {
+        return codeGenCompoundAssignToVar(cx, startBlock, env, lValue, expr, op, pos);
+    }
+    else {
+        return codeGenCompoundAssignToMember(cx, startBlock, env, lValue, expr, op, pos);
+    }
+}
+
+
+function codeGenCompoundAssignToVar(CodeGenContext cx, bir:BasicBlock startBlock, Environment env, s:VarRefExpr lValue, s:Expr rexpr, s:BinaryArithmeticOp|s:BinaryBitwiseOp  op, err:Position pos) returns CodeGenError|StmtEffect {
+    s:Expr expr;
+    if op is s:BinaryArithmeticOp {
+        expr = { arithmeticOp: op, left: lValue, right: rexpr, pos: pos };
+    }
+    else {
+        expr = { bitwiseOp: <s:BinaryBitwiseOp> op, left: lValue, right: rexpr };
+    }
+    return codeGenAssignToVar(cx, startBlock, env, lValue.varName, expr);
+}
+
+function codeGenCompoundAssignToMember(CodeGenContext cx, bir:BasicBlock bb, Environment env, s:MemberAccessLExpr lValue, s:Expr rexpr, s:BinaryArithmeticOp|s:BinaryBitwiseOp op, err:Position pos) returns CodeGenError|StmtEffect {
+    var { result: r0, block: block1 } = check codeGenExpr(cx, bb, env, check cx.foldExpr(env, lValue.container, ()));
+    bir:Register r3;
+    bir:Operand r1;
+    bir:Operand r4;
+    bir:BasicBlock block;
+    if r0 is bir:Register {
+        bir:BasicBlock block3;
+        if t:isSubtypeSimple(r0.semType, t:LIST) {
+            { result: r1, block: block3 } = check codeGenExprForInt(cx, block1, env, check cx.foldExpr(env, lValue.index, t:INT));
+            r3 = cx.createRegister(<t:UniformTypeBitSet>t:simpleArrayMemberType(cx.mod.env, r0.semType));
+            bir:ListGetInsn insn = { result: r3, list: r0, operand: <bir:IntOperand> r1, position: pos };
+            block3.insns.push(insn);
+            var { result: r2, block: block2 } = check codeGenExpr(cx, block3, env, check cx.foldExpr(env, rexpr, t:ANY));
+            if op is s:BinaryArithmeticOp {
+                {result: r4, block} = check codeGenArithmeticBinaryExpr(cx, block2, r3, r2, op, pos);
+            }
+            else {
+                {result: r4, block} = check codeGenBitwiseBinaryExpr(cx, block2, r3, <bir:IntOperand> r2, op);  
+            }
+            bir:ListSetInsn insn1 = { list: <bir:Register> r0, index: <bir:IntOperand> r1, operand: r4, position: lValue.pos };
+            block.insns.push(insn1);
+            return { block };
+        }
+        else {
+            return cx.semanticErr("can only apply member compound assignment to list", pos=pos);
```

Comment:
Deal with this error case at the beginning of the function without indenting everything below.